### PR TITLE
task DEV-4

### PR DESCRIPTION
**Pull request by AKhlebko**
Anton Khlebko task DEV-4
Elements are sorted by their names, don't know how to sort them by their attributes
Attributes are sorted in every instance by their full name. Comparison based on built in comparator

### CODE REVIEW

File: DEV-4/DEV-4/XMLparser.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
+﻿using System.Text.RegularExpressions;
+
+namespace TaskDEV4
+{
+    /// <summary>
+    /// Class's for parsing XML files
+    /// </summary>
+    class XMLparser
+    {
+
```
Comment (IgorKheidorov) [inline DEV-4/DEV-4/XMLparser.cs]: code style, no comment

File: DEV-4/DEV-4/XMLparser.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
+﻿using System.Text.RegularExpressions;
+
+namespace TaskDEV4
+{
+    /// <summary>
```
Comment (IgorKheidorov) [inline DEV-4/DEV-4/XMLparser.cs]: Bad comment, grammar

File: DEV-4/DEV-4/XmlAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace TaskDEV4
+{
+    /// <summary>
+    /// Class for storing elements attribues
+    /// </summary>
+    class XmlAttribute : IComparable<XmlAttribute>
+    {
```
Comment (IgorKheidorov) [inline DEV-4/DEV-4/XmlAttribute.cs]: Code style

File: DEV-4/DEV-4/EntryPoint.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            try
+            {
+                string xmlString = File.ReadAllText(@args[0]);
+                XMLparser parser = new XMLparser();
+                XmlElement rootElement = new XmlElement("Parsed XML file");
+                xmlString = parser.DeleteComments(xmlString);
+                parser.DepthXMLParse(rootElement, xmlString);
+                rootElement.Sort();
+                rootElement.PrintFromRootElement();
+            }
```
Comment (IgorKheidorov) [inline DEV-4/DEV-4/EntryPoint.cs]: Are you sure that it is the only exception type possible to be thrown?

Comment (AKhlebko) [reply DEV-4/DEV-4/EntryPoint.cs]: I didn't really make accent on exceptions. My bad

**Pull request merged**